include/ozo/io/binary_query.h

```diff
@@ -134,8 +134,12 @@ class binary_query {
         std::size_t offset = 0;
         hana::for_each(range, [&] (auto field) {
                 const auto size = result->lengths[field];
-                result->values[field] = size ? std::data(result->buffer) + offset : nullptr;
-                offset += size;
+                if (size && size != -1) {
```

Comment:
There is special constant type for it
```suggestion
                if (size && size != null_state_size) {
```